hs_file_types/views.py

```diff
@@ -1422,6 +1422,57 @@ def update_model_instance_metadata(request, file_type_id, **kwargs):
     return JsonResponse(ajax_response_data, status=status.HTTP_200_OK)
 
 
+@authorise_for_aggregation_edit(file_type="CSVLogicalFile")
+@login_required
+def update_csv_table_schema_metadata(request, file_type_id, **kwargs):
+    """Updates the tableSchema metadata field of the CSVFileMetadata object associated with the CSVLogicalFile
+    """
+
+    # Note: decorator 'authorise_for_aggregation_edit' sets the error_response key in kwargs
+    if 'error_response' in kwargs and kwargs['error_response']:
+        error_response = kwargs['error_response']
+        return JsonResponse(error_response, status=status.HTTP_400_BAD_REQUEST)
+
+    # Note: decorator 'authorise_for_aggregation_edit' sets the logical_file key in kwargs
+    logical_file = kwargs['logical_file']
+    metadata = logical_file.metadata
+    post_table_schema_data = request.POST.dict()
+    table_schema_model = metadata.get_table_schema_model()
+    for col_no, col in enumerate(table_schema_model.table.columns):
+        col.titles = post_table_schema_data[f"column-{col_no}-titles"].strip()
+        if col.titles == '':
+            col.titles = None
+        col.description = post_table_schema_data[f"column-{col_no}-description"].strip()
+        if col.description == '':
+            col.description = None
+        col.datatype = post_table_schema_data[f"column-{col_no}-datatype"].strip()
+
+    # it is not allowed to have some columns with titles and some without titles
+    columns_with_titles = [col.titles for col in table_schema_model.table.columns if col.titles]
+    if 0 < len(columns_with_titles) < len(table_schema_model.table.columns):
+        ajax_response_data = {'status': 'error', 'logical_file_type': logical_file.type_name(),
+                              'message': "Some column titles are missing"}
+        return JsonResponse(ajax_response_data, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
![image](https://github.com/user-attachments/assets/b3f6b970-896d-4305-a282-aa7a5b12439d)
I would revise this message to make it more clear what is wrong. Also the QUOTES in the message are a bit odd...

Reply:
Removed the quotes and tried to improve the error message.